zigpy/config/__init__.py

```diff
@@ -124,3 +126,7 @@
 CONFIG_SCHEMA = ZIGPY_SCHEMA.extend(
     {vol.Required(CONF_DEVICE): SCHEMA_DEVICE}, extra=vol.ALLOW_EXTRA
 )
+
+ENDPOINT_SCHEMA = vol.Schema(
+    {vol.Optional(CONF_ADDITIONAL_ENDPOINTS): SimpleDescriptor}
+)
```

Comment:
This schema isn't actually utilized. Instead of parsing the descriptor objects within `application.py`, move the parsing in here:

```python
        vol.Optional(
            CONF_NWK_VALIDATE_SETTINGS, default=CONF_NWK_VALIDATE_SETTINGS_DEFAULT
        ): cv_boolean,
        vol.Optional(CONF_ADDITIONAL_ENDPOINTS, default=[]): [
            lambda v: SimpleDescriptor(**v)
        ],

...
```

That way, `self.config[conf.CONF_ADDITIONAL_ENDPOINTS]` will always either be an empty list, or a list of `SimpleDescriptor` objects. The application will never have to parse them and you will receive a validation error if the config is invalid, before zigpy has a chance to start up.